aten/src/ATen/cuda/Exceptions.h

```diff
@@ -70,21 +71,26 @@ namespace at { namespace cuda { namespace solver {
 C10_EXPORT const char* cusolverGetErrorMessage(cusolverStatus_t status);
 }}} // namespace at::cuda::solver
 
-#define TORCH_CUSOLVER_CHECK(EXPR)                                              \
-  do {                                                                          \
-    cusolverStatus_t __err = EXPR;                                              \
-    if (__err == CUSOLVER_STATUS_EXECUTION_FAILED) {                            \
-      TORCH_CHECK(__err == CUSOLVER_STATUS_SUCCESS,                             \
-                  "cusolver error: ",                                           \
-                  at::cuda::solver::cusolverGetErrorMessage(__err),             \
-                  ", when calling `" #EXPR "`",                                 \
-                  ". This error may appear if the input matrix contains NaN."); \
-    } else {                                                                    \
-      TORCH_CHECK(__err == CUSOLVER_STATUS_SUCCESS,                             \
-                  "cusolver error: ",                                           \
-                  at::cuda::solver::cusolverGetErrorMessage(__err),             \
-                  ", when calling `" #EXPR "`");                                \
-    }                                                                           \
+#define TORCH_CUSOLVER_CHECK(EXPR)                                      \
+  do {                                                                  \
+    cusolverStatus_t __err = EXPR;                                      \
+    if ((CUSOLVER_VERSION < 11500 &&                                    \
```

Comment:
Could you clarify this conditions? Why these errors should be linalg error and not the others?

Reply:
I've added some comments that were already there in test_linalg: https://github.com/pytorch/pytorch/blob/master/test/test_linalg.py#L2940